scripts/SmartHistory.lua

```diff
@@ -50,46 +37,15 @@ mp.register_event('file-loaded', function()
 	end
 end)
 
-mp.register_event('playback-restart', function()
-	seconds = 0
-	time = math.floor(mp.get_property_number('time-pos') or 0)
-	seconds = seconds + time
-end)
-
-mp.observe_property('pause', 'bool', function(name, value)
-	if value then
-		if timer ~= nil then
-			timer:stop()
-		end
-		time = math.floor(mp.get_property_number('time-pos') or 0)
-		seconds = time
-		pause = true
-	else
-		if timer ~= nil then
-			timer:resume()
-		end
-		time = math.floor(mp.get_property_number('time-pos') or 0)
-		seconds = time
-		pause = false
-	end
-end)
-
-mp.register_event('end-file', function()
+mp.add_hook('on_unload', 50, function()
```

Comment:
Interesting, what is the 50 for in this line?

Reply:
This is `priority` argument.

> mp.add_hook(type, priority, fn)

https://mpv.io/manual/master/#lua-scripting-mp-add-hook(type,-priority,-fn)

Reply:
Thanks for the clarification, I'll test it out and then merge